lib/samplers/sampler.js

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2025 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @private
+ * @interface
+ */
+class Sampler {
+  /**
+   * Sets `priority` and `sampled` on the transaction
+   * in respect to this sampler's decision.
+   * @param {Transaction} transaction the transaction to update
+   */
+  applySamplingDecision(transaction) {
```

Comment:
I'm unclear here. We've conflated a sampler with a sampler. Meaning previous samplers like adaptive sampler were called with `shouldSample`